client/windows/QuickRecruitmentWindow.cpp

```diff
@@ -53,8 +53,10 @@ void QuickRecruitmentWindow::setCreaturePurhaseCards()
 	{
 		if(!town->town->creatures.at(i).empty() && !town->creatures.at(i).second.empty() && town->creatures[i].first)
 		{
-			CreatureID crid = town->creatures[i].second[town->creatures[i].second.size() - 1];
-			cards.push_back(std::make_shared<CreaturePurhaseCard>(VLC->creh->creatures[crid], position, town->creatures[i].first, this));
+			std::vector<CreatureID> crids;
+			for(auto j : town->creatures[i].second)
+				crids.push_back(j);
+			cards.push_back(std::make_shared<CreaturePurhaseCard>(crids, position, town->creatures[i].first, this));
```

Comment:
No need to copy `town->creatures[i].second`, pass constant reference to CreaturePurhaseCard ctor.

Reply:
heheh, what a stupid mistake, fixed